src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -689,24 +673,36 @@ void BasePlaylistFeature::slotTrackSelected(TrackPointer pTrack) {
     }
     m_playlistDao.getPlaylistsTrackIsIn(trackId, &m_playlistsSelectedTrackIsIn);
 
-    // Set all playlists the track is in bold (or if there is no track selected,
-    // clear all the bolding).
     for (int row = 0; row < m_childModel.rowCount(); ++row) {
         QModelIndex index = m_childModel.index(row, 0);
         TreeItem* pTreeItem = m_childModel.getItem(index);
         DEBUG_ASSERT(pTreeItem != nullptr);
-        if (!pTreeItem->hasChildren()) { // leaf node
-            bool ok;
-            int playlistId = pTreeItem->getData().toInt(&ok);
-            VERIFY_OR_DEBUG_ASSERT(ok) {
-                continue;
+        markTreeItem(pTreeItem);
+    }
+
+    m_childModel.triggerRepaint();
+}
+
+void BasePlaylistFeature::markTreeItem(TreeItem* pTreeItem) {
+    bool ok;
+    int playlistId = pTreeItem->getData().toInt(&ok);
+    if(ok) {
+        bool shouldBold = m_playlistsSelectedTrackIsIn.contains(playlistId);
+        pTreeItem->setBold(shouldBold);
+        if (shouldBold && pTreeItem->hasParent()) {
+            TreeItem* item = pTreeItem;
+            while ((item = item->parent())) {
```

Comment:
They are not redundant, because the compiler will generate a warning without extra parentheses when doing an assignment inside a loop condition. The root issue is that assignments inside loop conditions are confusing and should be avoided IMHO.